### PR TITLE
Merge latest cloud into master

### PR DESCRIPTION
**Pull request by jwilander**

### CODE REVIEW

**Comment by jwilander** [comment]
Because we didn't do a merge commit the last couple times (it's disabled in the UI) the old commits are showing up. I'll do a manual merge commit when the PR is approved to fix it for the future.

@marianunez @mgdelacroix if you could take a look at the `ea7468b` commit to make sure it looks good (it's all been tested) that'd be great

File: store/sqlstore/upgrade.go [hunk 1]
```diff
@@ -940,31 +941,107 @@ func upgradeDatabaseToVersion529(sqlStore *SqlStore) {
...
+		)
+		return true
+	}
+
+	if sqlStore.DriverName() == model.DATABASE_DRIVER_POSTGRES {
+		if !sqlStore.IsVarchar(scIdInfo.DataType) || scIdInfo.CharMaximumLength != 300 {
+			return true
+		}
+	} else if sqlStore.DriverName() == model.DATABASE_DRIVER_MYSQL {
+		if scIdInfo.DataType != "text" {
```
Comment (marianunez) [inline store/sqlstore/upgrade.go:1037]: Do we want to check the length in this case as well to be consistent?

**Review by marianunez — APPROVED** [review]
Checked migration checker [commit](https://github.com/mattermost/mattermost-server/commit/ea7468be52e550271d2dce3c3955bf0a6fecf5c4)

LGTM

File: store/sqlstore/upgrade.go (hunk 1 above)
Comment (jwilander) [reply store/sqlstore/upgrade.go:1037]: There's no length for `text` columns in MySQL

Comment (marianunez) [reply store/sqlstore/upgrade.go:1037]: 🤦‍♀️

**Comment by marianunez** [comment]
I believe Miguel is OOO this week, so we may want to swap out the reviewer with possibly someone else in platform?

**Comment by jwilander** [comment]
@agnivade this is mainly a merge commit (please excuse the number of commits that was the result of not merge committing and squashing previously) but if you could give https://github.com/mattermost/mattermost-server/commit/ea7468be52e550271d2dce3c3955bf0a6fecf5c4 a review that would be great

**Comment by jwilander** [comment]
There's an issue with the merge causing some tests to fail. I'm looking into it.

**Comment by jwilander** [comment]
Tests should be fixed now.

**Comment by agnivade** [comment]
@jwilander - The code could use some more refactoring. The `GetColumnInfo` and the subsequent check should be in a single method like `DoesColumnExist`. Perhaps something like `IsColumnOfType`. But I guess you just wanted a sanity check and not a PR review right?

Looks good to me otherwise.

**Comment by jwilander** [comment]
Yes, looking mainly for a sanity check but the feedback is always appreciated. The special cloud migration code should be temporary and go away once we land on our official cloud friendly schema migration strategy.

**Review by agnivade — APPROVED** [review]

**Pull request merged**